knotx-core/src/main/java/io/knotx/dataobjects/KnotContext.java

```diff
@@ -91,6 +94,15 @@ public KnotContext setFragments(List<Fragment> fragments) {
     return this;
   }
 
+  public List<FileData> getFilesData() {
```

Comment:
Vert.x keeps information about files in defined structured but can't transformed them into json which is required by knot.x, That is why new structure is created. And it is not API so we don't need generic